lib/jekyll-github-metadata/edit-link-tag.rb

```diff
@@ -72,6 +71,14 @@ def parts
         memoize_conditionally { [repository_url, "edit/", branch, source_path, page_path] }
       end
 
+      def page_path
+        return page["path"] unless page["paginated"]
+
+        site.pages.dup.select do |page|
```

Comment:
There's no need to use `.dup` here.
`Array#select` does not mutate the original array.

Additionally, since you're not using the *filtered* result of `Array#select`, you might as well iterate directly
```suggestion
        site.pages.each do |page|
```

Reply:
Fair enough, I’d just lifted that from jekyll-paginate so wasn't sure, but I see [in an unreleased commit they had done the same](https://github.com/jekyll/jekyll-paginate/commit/70efd9695cb49157fbc74509d5eb2e48ee266252). I had figured `.dup` would reduce the number of pages iterated over to find the right path?

Reply:
> I had figured `.dup` would reduce the number of pages iterated over to find the right path?

Nope. `.dup` simply allocates a *duplicate* copy of the original array. Another *duplicate* gets created by the subsequent `.select`.

Reply:
Thanks for the explanation — having only seen `.clone` in the past, I had mistaken `.dup` for something intrinsic to Jekyll.